cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/conversion/WritableConversions.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch.conversion;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.etl.batch.StructuredRecordWritable;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.io.BooleanWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.DoubleWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Functions to convert common classes to their WritableComparable equivalent and vice versa.
+ */
+public class WritableConversions {
+  // classname -> function for functions that transform common classes that should be transformed into classes usable
+  // as mapper keys or values.
+  private static final Map<String, WritableConversion<?, ?>> conversions =
+    ImmutableMap.<String, WritableConversion<?, ?>>builder()
```

Comment:
Constructing the map like this is a bit hard to read. I think it's better to use a static initializer block. Also, can the `WritableConversion` class have method like `toWritable` and `fromWritable` (or `wrap`, `unwrap`) directly instead of returning a function for doing so?